### DISCUSSION TITLE
Can't link skins.so

### DISCUSSION THREAD

**Discussion by y0shir** [audacious-media-player/audacious#131] [Q&A] [answered]
I am trying to apply patch how classic skins work and run into a bit of trouble.
I have run both `autogen.sh` and `./configure`, both of which gave no errors. Afterwards I have run `make`, which gave me following error:
```
/usr/bin/ld: menus.plugin.o: в функции «position_menu(_GtkMenu*, int*, int*, int*, void*)»:
/home/user/Документы/programming/audacious/audacious-plugins-audacious-plugins-4.3.1/audacious-plugins-audacious-plugins-4.3.1/src/skins/menus.cc:298:(.text+0x115): undefined reference to `audgui_get_monitor_geometry(_GdkScreen*, int, int, _GdkRectangle*)'
collect2: error: ld returned 1 exit status
Failed to link skins.so!
make[5]: *** [../../buildsys.mk:230: skins.so] Ошибка 1
make[4]: *** [../../buildsys.mk:135: all] Ошибка 2
make[3]: *** [../buildsys.mk:145: skins] Ошибка 2
make[2]: *** [../buildsys.mk:133: all] Ошибка 2
make[1]: *** [buildsys.mk:145: src] Ошибка 2
make: *** [buildsys.mk:133: all] Ошибка 2
```
After searching a bit I found that needed function is defined in libaudgui, so I checked libaudgui.so for symbols and it's there
```
/usr/lib/x86_64-linux-gnu$ nm -gD libaudgui.so |grep get_monitor
                 U gdk_display_get_monitor_at_point
00000000000221b0 T _Z27audgui_get_monitor_geometryP10_GdkScreeniiP20_cairo_rectangle_int
```
Althought it seems like second struct is cairo rectangle instead of gdk one.

Do I not get something or is there some hidden incompatibility? I am using debian testing and audacious  4.3.1 (Debian package)

**Comment by radioactiveman** [comment, marked as answer]
Debian Testing enables GTK3 for their Audacious packages. Therefore you need to do the same with `./configure --enable-gtk3`.
